File: src/main/java/com/leon/bugreport/extensions/BugReportPair.java
package com.leon.bugreport.extensions;

import org.jetbrains.annotations.Contract;
import org.jetbrains.annotations.NotNull;

public class BugReportPair<L, R> {
	private final L first;
	private final R second;

	public BugReportPair(L first, R second) {
		this.first = first;
		this.second = second;
	}

	public L getFirst() {
		return first;
	}

	public R getSecond() {
		return second;
	}

	@Contract(value = "_, _ -> new", pure = true)
	public static <L, R> @NotNull BugReportPair<L, R> of(L first, R second) {
		return new BugReportPair<>(first, second);
	}
}

File: src/main/java/com/bugreportmc/bugreport/extensions/BugReportPair.kt
package com.bugreportmc.bugreport.extensions

import org.jetbrains.annotations.Contract

class BugReportPair<L, R>(val first: L, val second: R) {
	companion object {
		@Contract(value = "_, _ -> new", pure = true)
		fun <L, R> of(first: L, second: R): BugReportPair<L, R> {
			return BugReportPair(first, second)
		}
	}
}
